lib/runner.ex

```diff
@@ -1,31 +1,31 @@
 defmodule Runner do
-  @modules [
-    Equalities,
-    Strings,
-    Numbers,
-    Atoms,
-    Tuples,
-    Lists,
-    KeywordLists,
-    Maps,
-    MapSets,
-    Structs,
-    Sigils,
-    PatternMatching,
-    Functions,
-    Enums,
-    Processes,
-    Tasks,
-    Agents,
-    Protocols,
-  ]
-
-  def koan?(koan), do: Enum.member?(@modules, koan)
-
-  def modules, do: @modules
+
+  def koan?(koan) do
+    Keyword.has_key?(koan.__info__(:exports), :all_koans)
+  end
+
+  def modules do
+    {:ok, modules} = :application.get_key(:elixir_koans, :modules)
+
+    modules
+    |> Stream.map(&(&1.module_info |> get_in([:compile, :source])))
+    |> Stream.map(&to_string/1)  # Paths are charlists
+    |> Stream.zip(modules)
+    |> Stream.filter(fn {_path, mod} -> koan?(mod) end)
+    |> Stream.map(fn {path, mod} -> {path_to_number(path), mod} end)
+    |> Enum.sort_by(fn {number, _mod} -> number end)
+    |> Enum.map(fn {_number, mod} -> mod end)
+  end
+
+  @koan_path_pattern ~r/lib\/koans\/(\d+)_\w+.ex$/
```

Comment:
Oh... I didn't thought about that. I can try it on a windows machine on Monday. If it does fail we can just ignore the entire path and match against `(\d+)_\w+.ex$`. With the `koan?` check this should be sufficient.